### PR TITLE
Promote Page Reorganized

### PR DESCRIPTION
**Pull request by dmcavoy**
Updating the promote page. Adding new sections for code.org impact and connecting to the advocacy site. Also updating the layout and removing a couple sections. Details below

# New Sections:

## Impact Stats:
screen shot 2019-02-06 at 8 45 57 am

## Advocacy Section:
screen shot 2019-02-06 at 8 46 15 am

# Updated Sections

## Updated Map Styles:
screen shot 2019-02-06 at 8 46 05 am
screen shot 2019-02-06 at 8 46 11 am

### Note: This update also updates the Advocacy.code.org home page
screen shot 2019-02-06 at 8 45 36 am

## New More Stats Button
screen shot 2019-02-06 at 8 45 52 am

# Removed:

## Promote Tools

screen shot 2019-02-04 at 4 04 32 pm

[Spec](https://docs.google.com/document/d/1rCOPhE2zBszuiKeNtBW5iAxzv8XogrWQFi6HIMicV_4/edit)

### CODE REVIEW

File: pegasus/sites.v3/code.org/views/stats_advocate_locally.haml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+%link{:href=>"/css/advocate-locally.css", :rel=>"stylesheet"}
+
+.join-coalition-wrapper
+  %img.left-side-image{src:"/images/promote/2018_HoC-463.jpg"}
```
Comment (breville) [inline pegasus/sites.v3/code.org/views/stats_advocate_locally.haml]: We should give this image a simpler and more relevant image filename.  

Also, in the screenshot the image is squished.. you can use `fit-/fill-` in the URL to crop it.  (See https://github.com/code-dot-org/code-dot-org/blob/8990f16a29d8a1cf9dba0ffb220bd424e8542810/lib/cdo/pegasus/graphics.rb)

File: pegasus/sites.v3/code.org/public/css/advocate-locally.css [hunk 2]
```diff
@@ -0,0 +1,47 @@
+.join-coalition-wrapper{
```
Comment (breville) [inline pegasus/sites.v3/code.org/public/css/advocate-locally.css]: nit: this file should have 2-space indents, and spaces before braces.

File: pegasus/sites.v3/code.org/views/stats_advocate_locally.haml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  .right
+    %p.title
+      Advocate for State and Federal Policy
+    %p.subheading
+      Together, we can make computer science education available to every student! Help us grow the movement to make Computer Science a fundamental part of the K-12 education system.
+    %a{href: "https://advocacy.code.org"}
+      %button
+        Code.org Advocacy Coalition
+
+
```
Comment (breville) [inline pegasus/sites.v3/code.org/views/stats_advocate_locally.haml]: nit: extra blank line

File: pegasus/sites.v3/code.org/public/css/advocate-locally.css [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  font-size: 26px;
+  padding-bottom: 10px;
+  color: white;
+}
+
+.join-coalition-wrapper .right .subheading {
+  font-size: 16px;
+}
+
+@media screen and (max-width: 971px)
```
Comment (maddiedierker) [inline pegasus/sites.v3/code.org/public/css/advocate-locally.css]: nit: put the opening bracket (`{`) on this line

File: pegasus/sites.v3/code.org/public/css/common.css [hunk 5]
```diff
@@ -471,46 +471,57 @@ button {
   margin: 15px 0 0 20px;
 }
 
-/* Stats carousel */
-#stats-carousel {
+/* Stats carousels */
+#stats-carousel,
+#stats-impact-carousel{
```
Comment (maddiedierker) [inline pegasus/sites.v3/code.org/public/css/common.css]:
```suggestion
#stats-impact-carousel {
```

File: pegasus/sites.v3/code.org/public/css/common.css [hunk 6]
```diff
@@ -471,46 +471,57 @@ button {
...
+#stats-impact-carousel .slide img {
   width: 100%;
   height: 100%;
   display: block;
 }
 #stats-carousel .slide-prev,
-#stats-carousel .slide-next {
+#stats-carousel .slide-next,
+#stats-impact-carousel .slide-prev,
+#stats-impact-carousel .slide-next{
```
Comment (maddiedierker) [inline pegasus/sites.v3/code.org/public/css/common.css]:
```suggestion
#stats-impact-carousel .slide-next {
```

File: pegasus/sites.v3/code.org/public/css/common.css [hunk 7]
```diff
@@ -471,46 +471,57 @@ button {
...
   position: absolute;
   top: 0;
   width: 40px;
   height: 100%;
   background-color: #f7f7f7;
 }
-#stats-carousel .slide-prev {
+#stats-carousel .slide-prev,
+#stats-impact-carousel .slide-prev{
```
Comment (maddiedierker) [inline pegasus/sites.v3/code.org/public/css/common.css]:
```suggestion
#stats-impact-carousel .slide-prev {
```

File: pegasus/sites.v3/code.org/public/css/common.css [hunk 8]
```diff
@@ -471,46 +471,57 @@ button {
...
   left: 0;
 }
-#stats-carousel .slide-next {
+#stats-carousel .slide-next,
+#stats-impact-carousel .slide-next{
```
Comment (maddiedierker) [inline pegasus/sites.v3/code.org/public/css/common.css]:
```suggestion
#stats-impact-carousel .slide-next {
```

File: pegasus/sites.v3/code.org/public/css/common.css [hunk 9]
```diff
@@ -528,8 +539,46 @@ button {
   font-weight: normal;
 }
 
+.impact-stats-link{
```
Comment (maddiedierker) [inline pegasus/sites.v3/code.org/public/css/common.css]:
```suggestion
.impact-stats-link {
```

File: pegasus/sites.v3/code.org/public/css/common.css [hunk 10]
```diff
@@ -528,8 +539,46 @@ button {
...
+  width: 100%;
+  color: white;
+  background-color: #7665a0;
+  margin-top: 20px;
+  margin-bottom: 20px;
+  overflow:hidden;
```
Comment (maddiedierker) [inline pegasus/sites.v3/code.org/public/css/common.css]:
```suggestion
  overflow: hidden;
```

File: pegasus/sites.v3/code.org/public/css/common.css [hunk 11]
```diff
@@ -528,8 +539,46 @@ button {
...
+  height: 50px;
+  font-family: 'Gotham 5r', sans-serif;
+}
+
+
+
 @media screen and (min-width: 970px) {
-  #stats-carousel .slide .slide-img {
+  #stats-carousel .slide .slide-img,
+  #stats-impact-carousel .slide .slide-img{
```
Comment (maddiedierker) [inline pegasus/sites.v3/code.org/public/css/common.css]:
```suggestion
  #stats-impact-carousel .slide .slide-img {
```

File: pegasus/sites.v3/code.org/public/css/common.css [hunk 12]
```diff
@@ -528,8 +539,46 @@ button {
...
+  float: right;
+}
+
+.impact-stats-link button {
+  margin-top: 0px;
+  width: 200px;
+  height: 50px;
+  font-family: 'Gotham 5r', sans-serif;
+}
+
```
Comment (maddiedierker) [inline pegasus/sites.v3/code.org/public/css/common.css:569]: nit: extra whitespace (should only be 1 line of whitespace between CSS rules)

File: pegasus/sites.v3/code.org/public/css/interactive-map.css [hunk 13]
```diff
@@ -3,11 +3,22 @@
   min-height: 425px;
   float: left;
   margin-right: -100px;
+  margin-top: 100px;
+}
+
+#right {
+  float: right;
+  overflow:hidden;
```
Comment (maddiedierker) [inline pegasus/sites.v3/code.org/public/css/interactive-map.css]:
```suggestion
  overflow: hidden;
```

File: pegasus/sites.v3/code.org/public/css/interactive-map.css [hunk 14]
```diff
@@ -3,11 +3,22 @@
...
 }
 
 #infobox {
   text-align: center;
   float: right;
+  background-color: rgb(231,232,234);
+  color: dimgrey;
+  padding: 20px;
+  text-align: center;
+  overflow:hidden;
```
Comment (maddiedierker) [inline pegasus/sites.v3/code.org/public/css/interactive-map.css]:
```suggestion
  overflow: hidden;
```

File: pegasus/sites.v3/code.org/public/css/interactive-map.css [hunk 15]
```diff
@@ -70,26 +66,47 @@
   width: 270px;
 }
 
-.fact i {
+.policy-title{
```
Comment (maddiedierker) [inline pegasus/sites.v3/code.org/public/css/interactive-map.css]:
```suggestion
.policy-title {
```

File: pegasus/sites.v3/code.org/public/css/interactive-map.css [hunk 16]
```diff
@@ -70,26 +66,47 @@
...
+  padding: 10px 0px;
+}
+
+.icon-fact .fa-times-circle{
```
Comment (maddiedierker) [inline pegasus/sites.v3/code.org/public/css/interactive-map.css]:
```suggestion
.icon-fact .fa-times-circle {
```

File: pegasus/sites.v3/code.org/public/css/interactive-map.css [hunk 17]
```diff
@@ -70,26 +66,47 @@
...
+  color: lightgray;
+  font-size: 48px;
+}
+
+.fa-circle{
```
Comment (maddiedierker) [inline pegasus/sites.v3/code.org/public/css/interactive-map.css]:
```suggestion
.fa-circle {
```

File: pegasus/sites.v3/code.org/public/css/interactive-map.css [hunk 18]
```diff
@@ -70,26 +66,47 @@
...
+  color: dimgray;
+  font-size: 48px;
+}
+
+.icon-fact .fa-check-circle{
```
Comment (maddiedierker) [inline pegasus/sites.v3/code.org/public/css/interactive-map.css]:
```suggestion
.icon-fact .fa-check-circle {
```

File: pegasus/sites.v3/code.org/public/css/interactive-map.css [hunk 19]
```diff
@@ -105,9 +122,15 @@
...
   font-size: 12px;
   line-height: 12px;
   text-align: center;
+  color: dimgrey;
   opacity: 0.8;
-  color: #BCC2C6;
+  padding-top: 10px;
 }
+
+#sources a{
```
Comment (maddiedierker) [inline pegasus/sites.v3/code.org/public/css/interactive-map.css]:
```suggestion
#sources a {
```

File: pegasus/sites.v3/code.org/views/interactive_map.haml [hunk 20]
```diff
@@ -4,16 +4,16 @@
...
 
 - advocacy_variation ||= false
-- fill_color = advocacy_variation ? '#0094a3' : '#3E484F'
-- stroke_color = advocacy_variation ? 'white' : '#7A7A7A'
-- hover_color = advocacy_variation ? '#59cad3' : '#59cad3'
-- click_color = advocacy_variation ? '#ffa400' : '#0094a3'
+- fill_color = '#0094a3'
+- stroke_color = 'white'
+- hover_color = '#59cad3'
+- click_color =  '#ffa400'
```
Comment (maddiedierker) [inline pegasus/sites.v3/code.org/views/interactive_map.haml]:
```suggestion
- click_color = '#ffa400'
```

**Review by maddiedierker — APPROVED** [review]
functionality looks great! i'm sorry in advance for all of the style nit comments 😄

**Comment by breville** [comment]
Some additional observations:

- When on mobile width, the `<h1>Computer Science Education Stats</h1>` heading is hardcoded to `800px` width which makes the page scrollable sideways when it shouldn't be.  This is likely an existing issue, but it would be great to fix it here.
- When on desktop width, the first impact slide is squished.. you can see that the pie circle isn't a proper circle.  We should check that all slides are showing at the proper aspect ratio.
- The first impact slide has a large body of purple that makes the purple dots basically impossible to see against it.  It might be worth investigating a design solution to this.
- Do we want the impact slides to have any of the same buttons/links underneath as the slides above them have?
- This change regresses http://localhost-advocacy.code.org:3000, notably the side box for the map.
e.g. for Ohio it should look like this:
screenshot 2019-02-04 14 44 41
But looks like this:
screenshot 2019-02-04 14 44 46
Presumably, we also want the side box to have that first appearance on /promote as well, since we're adopting the style from advocacy.code.org's map.  (This also means we can remove any legacy CSS/HAML used only for the old style.)

**Comment by dmcavoy** [comment]
* When on mobile width, the `<h1>Computer Science Education Stats</h1>` heading is hardcoded to `800px` width which makes the page scrollable sideways when it shouldn't be.  This is likely an existing issue, but it would be great to fix it here.

_I removed the hard coded width for the h1 headers. @breville was there anything else that needed to happen to make this work? it looks like the scroll bar still pops up at the smallest screen size_

* When on desktop width, the first impact slide is squished.. you can see that the pie circle isn't a proper circle.  We should check that all slides are showing at the proper aspect ratio.

_I'll look into this._

* The first impact slide has a large body of purple that makes the purple dots basically impossible to see against it.  It might be worth investigating a design solution to this.

_Ill ask Mark._

* Do we want the impact slides to have any of the same buttons/links underneath as the slides above them have?

_No Marina and Mark did not want the buttons underneath but instead the purple Want to see more data? area_

* This change regresses http://localhost-advocacy.code.org:3000, notably the side box for the map.
  e.g. for Ohio it should look like this:
  screenshot 2019-02-04 14 44 41
  But looks like this:
  screenshot 2019-02-04 14 44 46
  Presumably, we also want the side box to have that first appearance on /promote as well, since we're adopting the style from advocacy.code.org's map.  (This also means we can remove any legacy CSS/HAML used only for the old style.)

_I am asking Mark to clarify but the new style in the info box on the promote page is on purpose. I think Mark was considering doing it for both pages although he was probably going to leave the white background on advocacy. More info later_

**Review by breville — APPROVED** [review]
It would be great to update the screenshots in this PR for the record.

Thanks for working through this substantial change!

**Pull request merged**